test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -106,31 +107,31 @@ public static MockTransportService createNewService(Settings settings, Version v
     }
 
     /**
-     * Some tests use MockTransportService to do network based testing. Yet, we run tests in multiple JVMs that means
-     * concurrent tests could claim port that another JVM just released and if that test tries to simulate a disconnect it might
-     * be smart enough to re-connect depending on what is tested. To reduce the risk, since this is very hard to debug we use
-     * a different default port range per JVM unless the incoming settings override it
-     * use a non-default base port otherwise some cluster in this JVM might reuse a port
-     */
-    private static int getBasePort() {
-        final int basePort = 10300 + (ESTestCase.TEST_WORKER_VM * 100);
-        if (basePort < 10300 || basePort >= 65000) {
-            // to ensure we don't get illegal ports above 65536 in the getPortRange method
-            throw new AssertionError("Expected basePort to be between 10300 and 65000 but was " + basePort);
-        }
-        return basePort;
-    }
-
-    /**
-     * Returns a unique port range for this JVM starting from the computed base port (see {@link #getBasePort()})
+     * Returns a unique port range for this JVM starting from the computed base port
      */
     public static String getPortRange() {
-        int basePort = getBasePort();
+        // some tests use MockTransportService to do network based testing. Yet, we run tests in multiple JVMs that means
+        // concurrent tests could claim port that another JVM just released and if that test tries to simulate a disconnect it might
+        // be smart enough to re-connect depending on what is tested. To reduce the risk, since this is very hard to debug we use
+        // a different default port range per JVM unless the incoming settings override it
+        // use a non-default base port otherwise some cluster in this JVM might reuse a port
+
+        // We rely on Gradle implementation details here, the worker IDs are long values incremented by one  for the
+        // lifespan of the daemon this means that they can get larger than the allowed port range.
+        // Ephemeral ports on Linux start at 32768 so we modulo to make sure that we don't exceed that.
+        // This is safe as long as we have fewer than 224 Gradle workers running in parallel
+        // See also: https://github.com/elastic/elasticsearch/issues/44134
+        final int startAt = Math.round(
+            RandomizedTest.systemPropertyAsLong(ESTestCase.TEST_WORKER_VM_ID, 0)
```

Comment:
we can use `Math.floorMod(long, int)` here instead that returns an int. This means that there's no need for `Math.round`.
Also we should check that the long returned by `RandomizedTest.systemPropertyAsLong(ESTestCase.TEST_WORKER_VM_ID, 0)` is non-negative.

Reply:
This is actually needed because `systemPropertyAsLong` actually returns a `float`, seems like a bug in randomized testing, so `Math.floorMod` won't work.

Reply:
you're using the value as system property, not the actual system property. This means that this will always return 0 at the moment. Sounds like we need a test here ;)

I think it would be simpler to write something like the following:

```
final int startAt = ESTestCase.TEST_WORKER_VM_ID.equals(ESTestCase.DEFAULT_TEST_WORKER_ID) ?
            0 :
            Math.floorMod(Long.parseLong(ESTestCase.TEST_WORKER_VM_ID), 223);
```

Reply:
:man_facepalming: 
I fixed it and added a test